### ISSUE TITLE
weird/unexpected output format of makoctl list

### ISSUE THREAD

**Issue by dmxk062** [emersion/mako#473]
i don't really know when this started, but for a while now makoctl list has provided a different kind of output than usual:
```json
{
  "type": "aa{sv}",
  "data": [
    [
      [
        "app-name",
        {
          "type": "s",
          "data": "notify-send"
        },
        "app-icon",
        {
          "type": "s",
          "data": ""
        },
        "category",
        {
          "type": "s",
          "data": ""
        },
        "desktop-entry",
        {
          "type": "s",
          "data": ""
        },
        "summary",
        {
          "type": "s",
          "data": "test"
        },
        "body",
        {
          "type": "s",
          "data": ""
        },
        "id",
        {
          "type": "u",
          "data": 2
        },
        "urgency",
        {
          "type": "y",
          "data": 2
        },
        "actions",
        {
          "type": "a{ss}",
          "data": []
        }
      ]
    ]
  ]
}
```
instead of the previous and more logical:
```json
{
  "type": "aa{sv}",
  "data": [
    [
      [
        "app-name":{
          "type": "s",
          "data": "notify-send"
        },
        "app-icon":{
          "type": "s",
          "data": ""
        },
        "category":{
          "type": "s",
          "data": ""
        },
        "desktop-entry":{
          "type": "s",
          "data": ""
        },
        "summary":{
          "type": "s",
          "data": "test"
        },
        "body":{
          "type": "s",
          "data": ""
        },
        "id":{
          "type": "u",
          "data": 2
        },
        "urgency":{
          "type": "y",
          "data": 2
        },
        "actions":{
          "type": "a{ss}",
          "data": []
        }
      ]
    ]
  ]
}
```
i assume this is not intentional since this also breaks makoctl menu. i am not sure whether this is an issue in mako or busctl though, so please let me know if this is an issue in systemd. 
i am on arch with the newest version of mako and systemd.

**Comment by emersion** [comment]
Maybe a sd-bus change?

In any case, this JSON schema isn't great at all, would be way nicer to have something better.

**Comment by dmxk062** [comment]
i just checked using d-feet, and it seems to report everything correctly when i call the dbus method:
```json
[{'actions': {},
  'app-icon': '',
  'app-name': 'notify-send',
  'body': '',
  'category': '',
  'desktop-entry': '',
  'id': 5,
  'summary': 'test',
  'urgency': 2}]
```
so this might really be an issue in busctl. i'll try to hack smth together in python, maybe i can write a quick workaround for myself.

**Comment by emersion** [comment]
Ideally we should rewrite makoctl in plain C.

**Comment by dmxk062** [comment]
the latest systemd update on arch fixes this.

**Comment by jpalus** [comment]
Just in case anyone bumps into this:

- Upstream report: https://github.com/systemd/systemd/issues/28711
- Fix: https://github.com/systemd/systemd/pull/28713 (already included in `254.1`)
